java/core/src/main/java/sleeper/core/util/PollWithRetries.java

```diff
@@ -37,25 +41,56 @@ private PollWithRetries(long pollIntervalMillis, int maxPolls) {
         this.maxPolls = maxPolls;
     }
 
+    /**
+     * Creates an instance of this class.
+     *
+     * @param  pollIntervalMillis the poll interval in milliseconds
+     * @param  maxPolls           the maximum amount of polls
```

Comment:
Maybe "number of polls" would be slightly clearer than "amount of"? This also applies to the other static constructor.